upstairs/src/downstairs.rs

```diff
@@ -1961,17 +2036,51 @@ impl Downstairs {
     /// empty).
     ///
     /// # Panics
-    /// If `self.reconcile_current_work` is not `None`
+    /// If `self.reconcile_current_work` is not `None`, or if `self.reconcile`
+    /// is `None`.
     pub(crate) async fn send_next_reconciliation_req(&mut self) -> bool {
         assert!(self.reconcile_current_work.is_none());
+
+        let Some(ref mut reconcile) = self.reconcile else {
+            panic!("`self.reconcile` must be Some during reconciliation");
+        };
+
         let Some(mut next) = self.reconcile_task_list.pop_front() else {
             info!(self.log, "done with reconciliation");
             return true;
         };
+
+        reconcile.reconcile_task_list_index += 1;
+
+        info!(
+            self.log,
+            "reconciliation {}: on task {} of {}",
+            reconcile.id,
+            reconcile.reconcile_task_list_index,
+            self.reconcile_repair_needed,
+        );
+
+        #[cfg(feature = "omicron-build")]
+        {
+            // If building for production, then notify Nexus of reconciliation
+            // progress.
+            let reconcile_id = reconcile.id.clone();
```

Comment:
I believe this clone is unnecessary, because `Uuid` implements `Copy`.

You should do a Clippy pass, since it pops up a bunch of similar complaints.

Reply:
Weird, I always do `cargo clippy --all-targets`. fb4f36b addresses all the clippy errors on my system.

Reply:
You'll need to add`--features=omicron-build` as well to catch everything now.

Here's what I see with `cargo clippy -pcrucible --features=omicron-build`:
```
warning: using `clone` on type `Uuid` which implements the `Copy` trait
    --> upstairs/src/downstairs.rs:2067:32
     |
2067 |             let reconcile_id = reconcile.id.clone();
     |                                ^^^^^^^^^^^^^^^^^^^^ help: try removing the `clone` call: `reconcile.id`
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy
     = note: `#[warn(clippy::clone_on_copy)]` on by default

warning: use of `flat_map` with an identity function
    --> upstairs/src/downstairs.rs:3828:14
     |
3828 |             .flat_map(|x| x)
     |              ^^^^^^^^^^^^^^^ help: try: `flatten()`
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#flat_map_identity
     = note: `#[warn(clippy::flat_map_identity)]` on by default

warning: using `clone` on type `Option<SocketAddr>` which implements the `Copy` trait
    --> upstairs/src/downstairs.rs:3827:27
     |
3827 |             .map(|client| client.target_addr.clone())
     |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try removing the `clone` call: `client.target_addr`
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy

warning: using `clone` on type `TypedUuid<UpstairsRepairKind>` which implements the `Copy` trait
    --> upstairs/src/downstairs.rs:3893:40
     |
3893 | ...                   repair_id: repair_id.clone(),
     |                                  ^^^^^^^^^^^^^^^^^ help: try removing the `clone` call: `repair_id`
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy

warning: using `clone` on type `TypedUuid<UpstairsSessionKind>` which implements the `Copy` trait
    --> upstairs/src/downstairs.rs:3895:41
     |
3895 | ...                   session_id: session_id.clone(),
     |                                   ^^^^^^^^^^^^^^^^^^ help: try removing the `clone` call: `session_id`
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy

warning: using `clone` on type `TypedUuid<UpstairsRepairKind>` which implements the `Copy` trait
    --> upstairs/src/downstairs.rs:3977:40
     |
3977 | ...                   repair_id: repair_id.clone(),
     |                                  ^^^^^^^^^^^^^^^^^ help: try removing the `clone` call: `repair_id`
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy

warning: using `clone` on type `TypedUuid<UpstairsSessionKind>` which implements the `Copy` trait
    --> upstairs/src/downstairs.rs:3979:41
     |
3979 | ...                   session_id: session_id.clone(),
     |                                   ^^^^^^^^^^^^^^^^^^ help: try removing the `clone` call: `session_id`
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy

warning: using `clone` on type `TypedUuid<UpstairsRepairKind>` which implements the `Copy` trait
    --> upstairs/src/downstairs.rs:4127:40
     |
4127 | ...                   repair_id: repair_id.clone(),
     |                                  ^^^^^^^^^^^^^^^^^ help: try removing the `clone` call: `repair_id`
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy

warning: using `clone` on type `TypedUuid<UpstairsSessionKind>` which implements the `Copy` trait
    --> upstairs/src/downstairs.rs:4129:41
     |
4129 | ...                   session_id: session_id.clone(),
     |                                   ^^^^^^^^^^^^^^^^^^ help: try removing the `clone` call: `session_id`
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy

warning: using `clone` on type `TypedUuid<UpstairsRepairKind>` which implements the `Copy` trait
    --> upstairs/src/downstairs.rs:4217:40
     |
4217 | ...                   repair_id: repair_id.clone(),
     |                                  ^^^^^^^^^^^^^^^^^ help: try removing the `clone` call: `repair_id`
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy

warning: using `clone` on type `TypedUuid<UpstairsSessionKind>` which implements the `Copy` trait
    --> upstairs/src/downstairs.rs:4219:41
     |
4219 | ...                   session_id: session_id.clone(),
     |                                   ^^^^^^^^^^^^^^^^^^ help: try removing the `clone` call: `session_id`
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy

warning: using `clone` on type `Ipv6Addr` which implements the `Copy` trait
    --> upstairs/src/lib.rs:2133:29
     |
2133 |                 addr = Some(target_addr.ip().clone());
     |                             ^^^^^^^^^^^^^^^^^^^^^^^^ help: try dereferencing it: `*target_addr.ip()`
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy

warning: `crucible` (lib) generated 12 warnings (run `cargo clippy --fix --lib -p crucible` to apply 12 suggestions)
    Finished dev [unoptimized + debuginfo] target(s) in 0.27s
```

Reply:
yeah, --all-targets != --all-features. appeased in ea277d4